compute/instances/custom-machine-type/helper_class.go

```diff
@@ -0,0 +1,126 @@
+	CPUSeriesN1Limit       = TypeLimit{append([]int{1}, MakeRange(2, 97, 2)...), 922, 6656, true, 624 << 10}
+)
+
+type CustomMachineType struct {
+	zone, cpuSeries     string
+	memoryMb, coreCount int
+	typeLimit           TypeLimit
+}
+
+func (t CustomMachineType) check() error {
+	// Check whether the requested parameters are allowed. Find more information about limitations of custom machine
+	// types at: https://cloud.google.com/compute/docs/general-purpose-machines#custom_machine_types
+
+	// Check the number of cores
+	if len(t.typeLimit.allowedCores) > 0 && !containsInt(t.typeLimit.allowedCores, t.coreCount) {
+		return fmt.Errorf("invalid number of cores requested. Allowed number of cores for %v is: %v", t.cpuSeries, t.typeLimit.allowedCores)
+	}
+
+	// Memory must be a multiple of 256 MB
+	if t.memoryMb%256 != 0 {
+		return fmt.Errorf("requested memory must be a multiple of 256 MB")
+	}
+
+	// Check if the requested memory isn't too little
+	if t.memoryMb < t.coreCount*t.typeLimit.minMemPerCore {
+		return fmt.Errorf("requested memory is too low. Minimal memory for %v is %v MB per core", t.cpuSeries, t.typeLimit.minMemPerCore)
+	}
+
+	// Check if the requested memory isn't too much
+	if t.memoryMb > t.coreCount*t.typeLimit.maxMemPerCore && !t.typeLimit.allowExtraMemory {
+		return fmt.Errorf("requested memory is too large.. Maximum memory allowed for %v is %v MB per core", t.cpuSeries, t.typeLimit.maxMemPerCore)
+	}
+	if t.memoryMb > t.typeLimit.extraMemoryLimit && t.typeLimit.allowExtraMemory {
+		return fmt.Errorf("requested memory is too large.. Maximum memory allowed for %v is %v MB", t.cpuSeries, t.typeLimit.extraMemoryLimit)
+	}
+
+	return nil
+}
+
+func (t CustomMachineType) IsExtraMemoryUsed() bool {
+	return t.memoryMb > t.coreCount*t.typeLimit.maxMemPerCore
+}
+
+func (t CustomMachineType) String() string {
+	// Return the custom machine type in form of a string acceptable by Compute Engine API.
+
+	if containsString([]string{E2Small, E2Micro, E2Medium}, t.cpuSeries) {
+		return fmt.Sprintf("zones/%v/machineTypes/%v-%v", t.zone, t.cpuSeries, t.memoryMb)
+	}
+
+	if t.IsExtraMemoryUsed() {
+		return fmt.Sprintf("zones/%v/machineTypes/%v-%v-%v-ext", t.zone, t.cpuSeries, t.coreCount, t.memoryMb)
+	}
+
+	return fmt.Sprintf("zones/%v/machineTypes/%v-%v-%v", t.zone, t.cpuSeries, t.coreCount, t.memoryMb)
+}
+
+func (t CustomMachineType) ShortString() string {
+	// Return machine type in a format without the zone. For example, n2-custom-0-10240.
+	// This format is used to create instance templates.
+	ss := strings.Split(t.String(), "/")
+	return ss[len(ss)-1]
+}
+
+func createCustomMachineType(zone, cpuSeries string, memoryMb, coreCount int, tl TypeLimit) (*CustomMachineType, error) {
+	if containsString([]string{E2Small, E2Micro, E2Medium}, cpuSeries) {
+		coreCount = 2
+	}
+	cmt := CustomMachineType{zone, cpuSeries, memoryMb, coreCount, tl}
+	check := cmt.check()
```

Comment:
the variable here should be `err` since it's an error. Also, consider using this terser syntax:

``` golang
if err := cmt.check(); err != nil {
  // ... 
}
```